mibidata/tiff.py

```diff
@@ -88,10 +93,24 @@ def write(filename, image, sed=None, optical=None, ranges=None,
     if not isinstance(image, mi.MibiImage):
         raise ValueError('image must be a mibidata.mibi_image.MibiImage '
                          'instance.')
-    if image.coordinates is None or image.size is None:
-        raise ValueError('Image coordinates and size must not be None.')
-    if image.masses is None or image.targets is None:
-        raise ValueError('Image channels must contain both masses and targets.')
+    missing_required_metadata = []
+    for meta_attr in REQUIRED_METADATA_ATTRIBUTES:
+        if getattr(image, meta_attr) is None:
+            missing_required_metadata.append(meta_attr)
+    if missing_required_metadata:
+        if len(missing_required_metadata) == 1:
+            missing_metadata_error = (f'{missing_required_metadata[0]} is '
+                                      f'required and may not be None.')
+        else:
+            missing_fields = ''
+            for i, meta_attr in enumerate(missing_required_metadata):
+                if i == len(missing_required_metadata) - 1:
+                    missing_fields += f'{meta_attr} '
+                else:
+                    missing_fields += f'{meta_attr}, '
```

Comment:
Instead of looping through here, you can use `', '.join(missing_required_metadata)`